src/recovery/mod.rs

```diff
@@ -778,7 +797,7 @@ pub(crate) fn build_state_loading_dataflow<A: Allocate>(
 
                                 match update {
                                     StateUpdate::Upsert(state) => resume_state.insert(key, state),
-                                    StateUpdate::Reset => resume_state.remove(&key),
+                                    StateUpdate::Discard => resume_state.remove(&key),
```

Comment:
When i read this, I keep expecting the logic to be something like "Is there an updated state? great, add it to the stateful collection. Otherwise, discard it". Otherwise, I'm thinking there would be a 3rd possibility of no state update? I think it's the `Update` that's confusing me. Like it's either a `State::Upsert(state)` or a `State::Discard` but maybe the word `State` is too loaded to stand alone

Reply:
FWIW on the reading side (since this linked snippet is the loading during recovery), that situation is already handled: if we didn't write the update, then there wasn't an update, so don't do anything.